numpyro/contrib/autoguide/__init__.py

```diff
@@ -193,6 +195,14 @@ def __call__(self, *args, **kwargs):
 
         return result
 
+    def sample_posterior(self, rng, opt_state, *args, **kwargs):
+        sample_shape = kwargs.pop('sample_shape', ())
+        latent_size = np.size(self._init_latent)
+        transform = self.get_transform(opt_state)
```

Comment:
I'm not sure if this pattern will be general enough for the `AutoContinuous` family, e.g. in the case that we have data dependent guides. We may want to consider subclassing this into `AutoTransformed` or something. On the other hand, I have mixed feelings about having deeply nested class hierarchies. 

I'm fine leaving this in for now, but the interface issues will be worth clearing once we have a few more examples.

Reply:
You are right. I have not considered data dependent guide. Let's resort this when we have a concrete example.